homeassistant/components/squeezebox/browse_media.py

```diff
@@ -0,0 +1,170 @@
+"""Support for media browsing."""
+from homeassistant.components.media_player import BrowseError, BrowseMedia
+from homeassistant.components.media_player.const import (
+    MEDIA_CLASS_ALBUM,
+    MEDIA_CLASS_ARTIST,
+    MEDIA_CLASS_DIRECTORY,
+    MEDIA_CLASS_GENRE,
+    MEDIA_CLASS_PLAYLIST,
+    MEDIA_CLASS_TRACK,
+    MEDIA_TYPE_ALBUM,
+    MEDIA_TYPE_ARTIST,
+    MEDIA_TYPE_GENRE,
+    MEDIA_TYPE_PLAYLIST,
+    MEDIA_TYPE_TRACK,
+)
+
+LIBRARY = ["Artists", "Albums", "Tracks", "Playlists", "Genres"]
+
+MEDIA_TYPE_TO_SQUEEZEBOX = {
+    "Artists": "artists",
+    "Albums": "albums",
+    "Tracks": "titles",
+    "Playlists": "playlists",
+    "Genres": "genres",
+    MEDIA_TYPE_ALBUM: "album",
+    MEDIA_TYPE_ARTIST: "artist",
+    MEDIA_TYPE_TRACK: "title",
+    MEDIA_TYPE_PLAYLIST: "playlist",
+    MEDIA_TYPE_GENRE: "genre",
+}
+
+SQUEEZEBOX_ID_BY_TYPE = {
+    MEDIA_TYPE_ALBUM: "album_id",
+    MEDIA_TYPE_ARTIST: "artist_id",
+    MEDIA_TYPE_TRACK: "track_id",
+    MEDIA_TYPE_PLAYLIST: "playlist_id",
+    MEDIA_TYPE_GENRE: "genre_id",
+}
+
+CONTENT_TYPE_MEDIA_CLASS = {
+    "Artists": {"parent": MEDIA_CLASS_DIRECTORY, "children": MEDIA_CLASS_ARTIST},
+    "Albums": {"parent": MEDIA_CLASS_DIRECTORY, "children": MEDIA_CLASS_ALBUM},
+    "Tracks": {"parent": MEDIA_CLASS_DIRECTORY, "children": MEDIA_CLASS_TRACK},
+    "Playlists": {"parent": MEDIA_CLASS_DIRECTORY, "children": MEDIA_CLASS_PLAYLIST},
+    "Genres": {"parent": MEDIA_CLASS_DIRECTORY, "children": "genre"},
```

Comment:
Only standard media class constants are allowed as `media_class` or `children_media_class` attributes.

Reply:
Understood. Using `MEDIA_CLASS_DIRECTORY` in lieu of a specific class for genre.